core/src/main/java/org/apache/iceberg/BaseOverwriteFiles.java

```diff
@@ -48,6 +48,10 @@ protected OverwriteFiles self() {
 
   @Override
   protected String operation() {
+    if (deletesDataFiles() && !addsDataFiles()) {
+      return DataOperations.DELETE;
+    }
+
```

Comment:
This looks good, but does the inverse also apply here? I think it does: If OverwriteFiles only addsDataFiles we should produce an "append" operation.

Reply:
Also if we wanted to address that separately that's also good with me.

Reply:
makes sense, I've updated this in the second commit